Meetings/devel-call-2024-09-04.md

```diff
@@ -0,0 +1,27 @@
+# Meeting Minutes: SVSM Development Call (September 4th, 2024)
+
+## Attendees:
+
+Adam Dunlap, Diego Gonzalez Villalobos, Dionna Glaze, Hajira Zaman, Huibo Wang, James Bottomley, Joerg Roedel, Nicolai Stange, Oliver Steffen, Peter Fang, Rama AMD, Roy Hopkins, Stefano Garzarella, Tom Dohrmann, Vijay Dhanraj
+
+## Topics:
+
+### Early Attestation Process Document
+
+* KBS Identity: The group discussed the need for KBS identity verification. It was concluded that while it's not strictly necessary, the worst-case scenario without it is a denial of service.
+* Provisioning: The initial state creation should be done offline using a tool that can register the secret into the KBS.
+* TPM State: The TPM state should be considered as a secret that needs to be stored and recovered on subsequent boots.
+* KBS as Untrusted Entity: The KBS should be treated as a read-only, untrusted entity.
+
+#### Communication Channel and Protocol
+
+* HTTP Proxy: The group discussed using an HTTP proxy as a simple communication channel between SVSM and the KBS.
+* Attestation Protocol: The attestation protocol should be implemented in the SVSM.
+* Negotiation Phase: A negotiation phase between SVSM and the KBS is needed to establish communication parameters.
+* Attested TLS: Using attested TLS for communication between SVSM and the KBS is recommended.
```

Comment:
IIUC @deeglaze suggested `aTLS` between the proxy and SVSM if we implement our protocol between the two, and allow the attestation protocol implementation into the proxy.

So aTLS can be used to negotiate hash and crypto algorithms between SVSM and the proxy.
See https://docs.google.com/document/d/11ZsxP8jsviP3ddp9Hrn0rf6inttNw_Pbnz0psXlxlPs/edit?disco=AAABRuYNYHo

@deeglaze please correct me if I'm wrong.

Reply:
Also in this case, maybe we can add a link to the aTLS proposal: https://datatracker.ietf.org/doc/draft-fossati-tls-attestation/

Reply:
I don't know what crypto algorithms you're talking about. I'm thinking we'd just have a transport layer secure channel terminated in the proxy. The negotiation with whatever services can be GET requests for REST or something, but every service is going to have its own quirks. We ought not require that the VMM itself constructs the application requests to the connected server.

Reply:
> I'm thinking we'd just have a transport layer secure channel terminated in the proxy

And started from where?

Reply:
Let's move this discussion to the early attestation doc.

Reply:
Discussion moved to doc. Resolving.